packages/eslint-plugin/tests/docs-eslint-output-snapshots/prefer-readonly-parameter-types.shot

```diff
@@ -0,0 +1,358 @@
+
+// Incorrect because WrapperWithOther.otherProp is not readonly and not in the allowlist
+function fn3(arg: WrapperWithOther) {}
+             ~~~~~~~~~~~~~~~~~~~~~ Parameter should be a read only type.
+"
+`;
+
+exports[` 4`] = `
+"Incorrect
+Options: {"allow":["$",{"from":"file","name":"Foo"},{"from":"lib","name":"HTMLElement"},{"from":"package","name":"Bar","package":"bar-lib"}]}
+
+import { Foo } from 'some-lib';
+import { Bar } from 'incorrect-lib';
+
+interface HTMLElement {
+  prop: string;
+}
+
+// Incorrect because Foo is not a local type
+function fn1(arg: Foo) {}
+
+// Incorrect because HTMLElement is not from the default library
+function fn2(arg: HTMLElement) {}
+             ~~~~~~~~~~~~~~~~ Parameter should be a read only type.
+
+// Incorrect because Bar is not from "bar-lib"
+function fn3(arg: Bar) {}
+"
+`;
+
+exports[` 5`] = `
+"Correct
+Options: {"allow":["$",{"from":"file","name":"Foo"},{"from":"lib","name":"HTMLElement"},{"from":"package","name":"Bar","package":"bar-lib"}]}
+
+interface Foo {
+  prop: string;
+}
+
+interface Wrapper {
+  readonly sub: Foo;
+  readonly otherProp: string;
+}
+
+// Works because Foo is allowed
+function fn1(arg: Foo) {}
+
+// Works even when Foo is nested somewhere in the type, with other properties still being checked
+function fn2(arg: Wrapper) {}
+"
+`;
+
+exports[` 6`] = `
+"Correct
+Options: {"allow":["$",{"from":"file","name":"Foo"},{"from":"lib","name":"HTMLElement"},{"from":"package","name":"Bar","package":"bar-lib"}]}
+
+import { Bar } from 'bar-lib';
+
+interface Foo {
+  prop: string;
+}
+
+// Works because Foo is a local type
+function fn1(arg: Foo) {}
+
+// Works because HTMLElement is from the default library
+function fn2(arg: HTMLElement) {}
+
+// Works because Bar is from "bar-lib"
+function fn3(arg: Bar) {}
+"
+`;
+
+exports[` 7`] = `
+"Correct
+Options: {"allow":["$",{"from":"file","name":"Foo"},{"from":"lib","name":"HTMLElement"},{"from":"package","name":"Bar","package":"bar-lib"}]}
+
+import { Foo } from './foo';
+
+// Works because Foo is still a local type - it has to be in the same package
+function fn(arg: Foo) {}
+"
+`;
+
+exports[` 8`] = `
+"Incorrect
+Options: { "checkParameterProperties": true }
+
+class Foo {
+  constructor(private paramProp: string[]) {}
+                      ~~~~~~~~~~~~~~~~~~~ Parameter should be a read only type.
+}
+"
+`;
+
+exports[` 9`] = `
+"Correct
+Options: { "checkParameterProperties": true }
+
+class Foo {
+  constructor(private paramProp: readonly string[]) {}
+}
+"
+`;
+
+exports[` 10`] = `
+"Options: { "checkParameterProperties": false }
+
+class Foo {
+  constructor(
+    private paramProp1: string[],
+    private paramProp2: readonly string[],
+  ) {}
+}
+"
+`;
+
+exports[` 11`] = `
+"Incorrect
+Options: { "ignoreInferredTypes": true }
+
+import { acceptsCallback, CallbackOptions } from 'external-dependency';
+
+acceptsCallback((options: CallbackOptions) => {});
+"
+`;
```

Comment:
This _Invalid_ example doesn't actually contain any lint errors. That's because it's meant to illustrate how to deal with external dependencies (but we can't use multiple files in playground):

https://github.com/typescript-eslint/typescript-eslint/blob/98aa4f3d3c1522b0670dcaf9e614c8e128edf127/packages/eslint-plugin/docs/rules/prefer-readonly-parameter-types.md?plain=1#L295-L316

So... I'm not sure what's the best way to handle this...

How about skipping the lint errors count assertion for this particular code example by writing something in the code block meta? For example:

```
```ts option='{ "ignoreInferredTypes": true }' skipLintTests
```

Reply:
Yeah that makes sense, I'd be in favor of it.

Maybe ... `skipValidation`?

Reply:
Sounds great! Let's use `skipValidation`!